EIPS/eip-5453.md

```diff
@@ -0,0 +1,150 @@
+such as using a wrapper contract or proxy upgrade.
+
+## Reference Implementation
+
+```solidity
+
+// Per feature request in Macro https://github.com/ethereum/solidity/issues/10
+// Macro is not supported yet in
+// Implementer can just replicate the following expansion by hand until
+// The macro is supported in solidity.
+#define REMAIN_LENGTH extraData.length - LENGTH_OF_ENDORSEMENT;
+
+abstract contract ERC5453 {
+    modifier onlyEndorsed(bytes32 _msgDigest, bytes calldata _end/**orsement**/) {
+        require(_end.length == LENGTH_OF_ENDORSEMENT);
+        require(_end[_end.length - 8:] == MAGIC_WORLD);
+
+
+        // ERC-1271 prefix. See https://eips.ethereum.org/EIPS/eip-1271
+        string memory erc1271Prefix = "\x19Ethereum Signed Message:\n32";
+        bytes32 erc1271Hash = keccak256(abi.encodePacked(erc1271Prefix, _msgDigest));
+        address endorser = ecrecover(erc1271Hash,
+            uint8(_end[128]), // v
+            bytes32(_end[64:96]), // r
+            bytes32(_eSig[96:128]) // s
+        );
+        // _isEligibleEndorser is application specific.
+        require (_isEligibleEndorser(endorser), "Endorser is not eligible to transfer.");
+        _;
+    }
+}
+
+contract EndorsableERC721 is SomeERC721, ERC5453 {
+  function safeTransferFrom(
+    address from,
+    address to,
+    uint256 id,
+    bytes calldata extraData)
+
+    onlyEndorsed(  // used as modifier
+        keccak256(
+            abi.encodePacked(
+                from, to, id, amount,
+                extraData[:REMAIN_LENGTH], // first part of extraData is reserved for original use for extraData unendorsed.
+                extraData[REMAIN_LENGTH: REMAIN_LENGTH + 32], // nonce of endorsement for the {contract, endorser} combination
+                extraData[REMAIN_LENGTH + 32: REMAIN_LENGTH + 64], // valid_by for the endorsement
+            ),
+        data[REMAIN_LENGTH:] // the endorsement component
+    ) {
+    super.safeTransferFrom(from, to, id, extraData[:REMAIN_LENGTH]); // original
+  }
+}
+```
+
+## Security Considerations
+
+### Replay-attack
+Replay attack is a type of attack on cryptography authentication.
+In a narrower sense, it usually refers to is a type of attack that circumvent the crypto
+signature verification by reusing an existing signature for message being signed again.
+Any implementations relying on this ERC must realize that all smart endorsements
+described here are cryptography signatures and are *public* and can be obtained by anyone.
+They must forsee the possibility of replay of the transactions not only
+at the exactly deployment of the same smart contract, but also other deployment
+of similar smart contract source, or a version on another chainId etc.
+All the `nonce`, `valid_since`, `valid_by` were meant to restrict the surface of attack but
+might not eliminate risks of all such attacks, e.g. see **Phishing** for endorsement.
```

Comment:
```suggestion
A replay attack is a type of attack on cryptography authentication. In a narrow sense, it usually refers to a type of attack that circumvents the cryptographically signature verification by reusing an existing signature for a message being signed again. Any implementations relying on this EIP must realize that all smart endorsements described here are cryptographic signatures that are *public* and can be obtained by anyone. They must foresee the possibility of a replay of the transactions not only at the exact deployment of the same smart contract, but also other deployments of similar smart contracts, or of a version of the same contract on another `chainId`, or any other similar attack surfaces. The `nonce`, `valid_since`, and `valid_by` fields are meant to restrict the surface of attack but might not fully eliminate the risk of all such attacks, e.g. see the [Phishing](#phishing) section.
```
The wording change is optional (but recommended since it fixes the grammatical errors), but the formatting change is required.